agent/http.go

```diff
@@ -1120,3 +1120,15 @@ func (s *HTTPHandlers) parseFilter(req *http.Request, filter *string) {
 		*filter = other
 	}
 }
+
+func getPathSuffixUnescaped(path string, prefixToTrim string) (string, error) {
+	// The suffix may be URL-encoded, so attempt to decode
+	suffixRaw := strings.TrimPrefix(path, prefixToTrim)
+	suffixUnescaped, err := url.PathUnescape(suffixRaw)
+
+	if err != nil {
+		return suffixRaw, fmt.Errorf("failure in unescaping path param %q: %v", suffixRaw, err)
```

Comment:
```suggestion
		return suffixRaw, fmt.Errorf("cannot unescape path param %q: %w", suffixRaw, err)
```

Reply:
@kisunji : I agree with the wording change, but not sure I agree with the change from `%v` to `%w` for the error.

Per [this Golang page](https://go.dev/blog/go1.13-errors):
> ...wrapping an error makes that error part of your API. If you don’t want to commit to supporting that error as part of your API in the future, you shouldn’t wrap the error.

> It’s important to remember that whether you wrap or not, the error text will be the same. A person trying to understand the error will have the same information either way; the choice to wrap is about whether to give programs additional information so they can make more informed decisions, or to withhold that information to preserve an abstraction layer.

In this case, I don't think the caller needs to be able to able to unwrap the returned error to specifically look for the unescape error.

Reply:
That's totally fair! In this case it doesn't matter too much since our http handler ultimately consumes all the errors before returning plaintext or json to the client but certainly a good thing to keep in mind.